mobile/Components/AddCalendarButton.js

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useEffect } from 'react';
+import {
+  StyleSheet, TouchableOpacity, Text, Alert,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import * as Calendar from 'expo-calendar';
+
+const styles = StyleSheet.create({
+  calendarButton: {
+    backgroundColor: '#4C4C9B',
+    marginTop: 10,
+    padding: 8,
+    borderRadius: 5,
+  },
+  calendarButtonText: {
+    color: '#ffffff',
+    textAlign: 'center',
+  },
+});
+
+function AddToCalendarButton({ eventName, eventStartDate, eventEndDate }) {
+  const [calendarPermission, setCalendarPermission] = useState(null);
+  const [eventIdInCalendar, setEventIdInCalendar] = useState('');
```

Comment:
`eventIdInCalendar` is never used, but I assume you could use it as a safety check?